### ISSUE TITLE
Add parity controls

### ISSUE THREAD

**Issue by Strilanc** [Strilanc/Quirk#440]
When making distillation circuits, it is useful to be able to operate on Pauli product observables like Z*I*X*Y. Currently the shortest way to achieve this is to repeat the observable twice (once to get it onto a single qubit, once to restore). If there was a variant of the current controls where these variant controls were only satisfied if an odd number of them were individually satisfied, this problem would be fixed.

So basically:

- Augment the Controls class to include the parity concept
- Augment the control mask shader computation code to have a parity control argument
- Come up with some convention for drawing these things. Maybe a small half-boxed letter with "parity" written below it?
- Make sure things are working w.r.t. gates being disabled, displayed being controlled, etc.

**Comment by Strilanc** [comment]
Fixed in https://github.com/Strilanc/Quirk/pull/438/commits/2d0e00af6b8ad0956fff2982638946f62d2d623c
